### PR TITLE
GH-1773: Convert xmlinput to use IRIx

### LINKED ISSUE #1773: Update xmlinput (RDF/XML parsing) to use IRIx.
### Change

The work is described in [email Evolving RDF/XML support and ARP](https://lists.apache.org/thread/zom28q9dqc7h4mx8s4v6h28c6b9sx2qh)

In summary:
Package xmlinput use jena-iri, the IRI parser directly.
This issue is to switch xmlinput to use the IRIx abstraction layer for IRI support.

### Are you interested in contributing a pull request for this task?

Yes

### PR DESCRIPTION
**Pull request by afs**
GitHub issue resolved #1773.

Explanation:
https://lists.apache.org/thread/zom28q9dqc7h4mx8s4v6h28c6b9sx2qh

----

 - [x] Tests are included.
 - [ ] Documentation change and updates are provided for the [Apache Jena website](https://github.com/apache/jena-site/)
 - [x] Commits have been squashed to remove intermediate development commit messages.
 - [x] Key commit messages start with the issue number (GH-xxxx or JENA-xxxx)

By submitting this pull request, I acknowledge that I am making a contribution to the Apache Software Foundation under the terms and conditions of the [Contributor's Agreement](https://www.apache.org/licenses/contributor-agreements.html).

### CODE REVIEW

**Comment by namedgraph** [comment]
Good stuff! Will https://github.com/afs/x4ld/tree/main/iri4ld not be part of Jena?

**Comment by afs** [comment]
> Will https://github.com/afs/x4ld/tree/main/iri4ld not be part of Jena?

One step at a time. `iri4ld` changes all the error messages.

This PR is making the way IRIs are handled is consistent and pluggable everywhere. xmlinput is the last piece of that.

Currently, the system IRIx provider is jena-iri backed which, largely, gives compatibility. 

RIOT already has a differently configured IRIFactory and has for a long time.

Having been digging into the details for this PR, I found difference of behaviour between RIOT, RIOT using RDF/XML, and direct use of ARP to read RDF/XML (the latter when jena-arq is not on the classpath). The difference relates to `file:` URLs with a relative path. 

What was unspecified ([RFC 1738](https://www.rfc-editor.org/rfc/rfc1738.html)) is now defined ([RFC 8089](https://www.rfc-editor.org/rfc/rfc8089.html)).

This impacts `owl:imports` using `file:`. It depends whether the ontology is in RDF/XML or Turtle (any non RDF/XML syntax).

As an interested party, I hope you will test this PR.

**Review by rvesse — APPROVED** [review]
New implementation looks much cleaner with a much smaller package tree.

( Also 1900+ files added, at least there's test cases! 🤣 )

File: jena-core/src/main/java/org/apache/jena/rdfxml/xmlinput0/ARP0.java [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+ * Includes contributions from Simon Raboczi and Andrew Newman
+ */
+public class ARP0 implements ARPConfig
+{
+
+    final private RDFXMLParser arpf;
+
+/** Creates a new RDF Parser.
+ * Can parse one file at a time.
+ */
```
[the remark this answers is not in the document]
Comment (afs) [reply jena-core/src/main/java/org/apache/jena/rdfxml/xmlinput0/ARP0.java:74]: Yes. And weird code indentation. 

`xmlinput0`, the code up to 4.7.0, is minimally touched by this PR so there is a way to get back to it.

For the upgraded code, I've reformatted some code in areas the PR touches but not gone through the codebase.

File: jena-core/src/main/java/org/apache/jena/rdfxml/xmlinput/RDFXMLReader.java [hunk 2]
```diff
@@ -59,13 +62,28 @@ public class RDFXMLReader implements RDFReaderI, ARPErrorNumbers {
...
     public RDFXMLReader() {
         arpf = RDFXMLParser.create();
+        this.resolveInitialXmlBase = true;
+    }
+
+    /**
+     * Creates new JenaReader which does not resolve the XMLbase when parsing. This
+     * is the original behaviour of RDFXMLReader but that has not been exposed via
+     * RIOT; only directly creating an RDFXMLReader or running without RIOT
+     * (jena-arq) has had that behaviour.
```
[the remark this answers is not in the document]
Comment (afs) [reply jena-core/src/main/java/org/apache/jena/rdfxml/xmlinput/RDFXMLReader.java:79]: Changed to "(which is in jena-arq)" -- it's saying where RIOT come from.

File: jena-core/src/main/java/org/apache/jena/rdfxml/xmlinput0/ARP0.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (rvesse) [reply jena-core/src/main/java/org/apache/jena/rdfxml/xmlinput0/ARP0.java:74]: 👍

**Pull request merged**